api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -434,8 +445,15 @@ public Workspace saveAndCloneCohortsConceptSetsAndDataSets(Workspace from, Works
   public WorkspaceAccessLevel getWorkspaceAccessLevel(
       String workspaceNamespace, String workspaceId) {
     WorkspaceACL workspaceACL = fireCloudService.getWorkspaceAcl(workspaceNamespace, workspaceId);
+    Map<String, WorkspaceAccessEntry> workspaceAcls = workspaceACL.getAcl();
+    WorkspaceAccessEntry userAcl = workspaceAcls.get(userProvider.get().getEmail());
+    // Published / featured workspaces don't give ACLs to everyone per se, they give READER access
+    // to the registered user domain email, so we check for that one if the user doesn't have access
+    // themself.
+    WorkspaceAccessEntry actualAcl =
```

Comment:
This is also assuming the user is registered, which may not be true at this point.

Ugh - Sorry about this bug - this was really not even related to recent workspaces. This was introduced when we optimized the ACL check a few week's back - I was the reviewer and should have caught that. I'm not clear on how featured workspaces are currently functional, actually - this may have broader implications.

I think I'd prefer to just throw out this method, and replace it with the following: `getWorkspace(fields="accessLevel")` , this pushes the computation into Firecloud. I think this should be hopefully be equally performant.

Reply:
Sorry, do you mean replace this entire method with that call or replace the call to `fireCloudService` in this method with that call

Reply:
Sorry, I meant: leave the method signature as-is for `getWorkspaceAccessLevel()`, replace the implementation with a call to `getWorkspace()` instead.

Reply:
I'm still missing something here. I don't know how that change would handle either the user not being registered or the workspace being a featured workspace.

Reply:
Also - how would a user that is not registered be able to even look at featured workspaces?

Reply:
I went digging through Rawls. It looks like, under the hood, the thing that gets called when when you do `getWorkspace(fields="accessLevel")` is `getMaximumAccessLevel`, which has an [all caps warning](https://github.com/broadinstitute/rawls/blob/7d26a387cf0267cf19d68921047b915a5dada2f8/core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala#L314) to not use it for actual access decisions. Talking to cloud accounts team about what the actual best way to answer the question of 'should this person be allowed to see / touch this thing' is.

Reply:
OK, thanks for digging - would be curious what they say on that.

RE: registered group, it should return the derived access level, which is what this method wants.
RE: how is an unregistered user able to look at featured workspaces? Exactly, they aren't supposed to be able to. So this method should behave consistently with that; it may even be used in places to make the decision about whether a user can look at them. This method is fairly generic and central - it should not be assuming that an incoming user has already been checked for registered access.

Reply:
Also to expand on that first bullet which wasn't very clear: in the featured workspaces case, the registered tier group is a reader. Registered users are members of this group, so their effective access level should be reader. Rawls should be doing this when it returns the derived access level, though there may be a more direct way.

(Disclaimer for PR scope: we don't currently have any direct calls to Sam)

Reply:
Cloud Accounts said 'talk directly to SAM' - we should call [this endpoint](https://sam.dsde-prod.broadinstitute.org/#!/Resources/resourceAction) with the resourceId we get back from asking Rawls about the workspace and then ask about action 'read'. So that's going to blow this up a bit